### PR TITLE
ref(metrics): Change default rule id from 0 to -1

### PR DESCRIPTION
**Pull request by iambriccardo**
This PR changes the default rule id for any newly created rule on the frontend from 0 to -1. This has been done in order to accomodate for the new default uniform sampling rule that will be implemented on the backend side (look at this [PR](https://github.com/getsentry/sentry/pull/40034)).

### CODE REVIEW

File: src/sentry/api/endpoints/project_details.py [hunk 1]
```diff
@@ -131,9 +132,13 @@ def fix_rule_ids(project, raw_dynamic_sampling):
             rules = raw_dynamic_sampling.get("rules", [])
 
             for rule in rules:
-                rid = rule.get("id", 0)
+                # We are setting the unassigned id to -1. It used to be 0 but we are modifying the behavior as 0 is a
+                # valid id according to relay's rule validation which states the a rule id is an unsigned integer.
+                rid = rule.get("id", -1)
                 original_rule = original_rules_dict.get(rid)
-                if rid == 0 or original_rule is None:
+                # ToDo(ahmed): Temporarily allowing for 0 to be the unassigned rule id for backwards compatibility,
```
Comment (priscilawebdev) [inline src/sentry/api/endpoints/project_details.py:139]:
```suggestion
                # TODO(ahmed): Temporarily allowing for 0 to be the unassigned rule id for backward compatibility,
```

**Comment by priscilawebdev** [comment]
@iambriccardo shall we create a separate PR for frontend changes? :)

**Comment by priscilawebdev** [comment]
it seems that your unit tests are failing and in order for them to pass you will have to do a couple of updates, for instance, in this object https://github.com/getsentry/sentry/blob/4900487abdf976fa834c48b4512a1f05b6d74411/static/app/views/settings/project/server-side-sampling/modals/specificConditionsModal/index.spec.tsx#L45 . You will have to replace `0` with `-1`

**Comment by iambriccardo** [comment]
@priscilawebdev this PR has been rebased on the backend implementation from @ahmedetefy but now the branch has been deleted because merged into master I think.

**Comment by priscilawebdev** [comment]
I see @iambriccardo so you will probably have to rebase this one against master

**Review by matejminar — COMMENTED** [review]
Can we link the backend PR in the description of this PR, please?

**Review by priscilawebdev — APPROVED** [review]
lgtm

**Pull request merged**